partner_customer_supplier/models/res_partner.py

```diff
@@ -0,0 +1,60 @@
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+DOMAIN_SEARCH = {
+    ("=", False): ("=", 0),
+    ("=", True): (">=", 1),
+    ("!=", False): (">=", 1),
+    ("!=", True): ("=", 0),
+}
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    customer = fields.Boolean(
+        string="Is Customer",
+        compute="_compute_partner_is_customer",
+        inverse="_inverse_partner_is_customer",
+        search="_search_partner_is_customer",
+    )
+    supplier = fields.Boolean(
+        string="Is Vendor",
+        compute="_compute_partner_is_supplier",
+        inverse="_inverse_partner_is_supplier",
+        search="_search_partner_is_supplier",
+    )
+
+    @api.depends("customer_rank")
+    def _compute_partner_is_customer(self):
+        for partner in self:
+            partner.customer = partner.customer_rank > 0
+
+    def _inverse_partner_is_customer(self):
+        self.filtered(lambda p: not p.customer).write({"customer_rank": 0})
+        self.filtered(lambda p: p.customer and not p.customer_rank).write(
+            {"customer_rank": 1}
```

Comment:
Question:

Shouldn't be counting the current customer account moves posted?

I mean, the main focus to use a rank is to have the best customer at the top of the views (e.g. m2o)

But if the user set to False then to True it will be reset to 1 so it will be at the bottom of the list

Something like `move.read_group([domain_is_sale + posted + partner], fields=id:count, group_by=partner_id)`

WDYT?

Reply:
This is expected to me. If I manually unset a customer, it probably means I don't trust them anymore. If I reset it to True, trust needs to be re-earned :)

Reply:
@moylop260 

As I see it, if Odoo user remove a partner from the customers (set `customer_rank = 0`), maybe he has many overdue debts and the responsible do not want to continue selling to him, so if at some point he settles the debt and later wants to buy again, he will have to earn back the "trust" (starting with `customer_rank = 1`).

cc: @luisg123v @tomeyro

Reply:
Ok, good point

I even disagree but you have a point

So, resolved

We can re-discuss it after